### ISSUE TITLE
Enthought Python Distribution and Spyder DMG are not working well on Mac OS X

### ISSUE THREAD

**Issue by spyder-issues-migrator** [spyder-ide/spyder#1277]
_From [breed...@gmail.com](https://code.google.com/u/105348566576685241380/) on 2013-02-20T15:23:46Z_

Spyder Version:  2.2.0beta2
Python Version:  2.7.3
Qt Version:      4.8.2, PyQt4 (API v2) 4.9.4 on Darwin 

What steps will reproduce the problem?

 1. Install Spyder from the disk image (not macports)
2. Go to Preferences>Console>Advanced Settings and then reassign the "Path to Python Interpreter Executable Binary" to /Library/Frameworks/EPD64.framework/Versions/7.3/bin/python.
3. Restart Spyder What is the expected output? What do you see instead? Expected output at console upon Spyder restart: 
Python 2.7.3 |CUSTOM| (default, Apr 12 2012, 11:14:05) 
[GCC 4.0.1 (Apple Inc. build 5493)] on darwin 

Actual output:
Python 2.7.3 |CUSTOM| (default, Apr 12 2012, 11:14:05) 
[GCC 4.0.1 (Apple Inc. build 5493)] on darwin 
Type "help", "copyright", "credits" or "license" for more information. 
Traceback (most recent call last): 
  File 
"/Applications/Spyder.app/Contents/Resources/lib/python2.7/spyderlib/scient ific_startup.py", 
line 16, in \<module> 
    from pylab import *  #analysis:ignore 
  File 
"/Library/Frameworks/EPD64.framework/Versions/7.3/lib/python2.7/site-packag es/pylab.py", 
line 1, in \<module> 
    from matplotlib.pylab import * 
  File 
"/Applications/Spyder.app/Contents/Resources/lib/python2.7/matplotlib/pylab .py", 
line 265, in \<module> 
    from matplotlib.pyplot import * 
  File 
"/Applications/Spyder.app/Contents/Resources/lib/python2.7/matplotlib/pyplo t.py", 
line 97, in \<module> 
    _backend_mod, new_figure_manager, draw_if_interactive, _show = 
pylab_setup() 
  File 
"/Applications/Spyder.app/Contents/Resources/lib/python2.7/matplotlib/backe nds/__init__.py", 
line 25, in pylab_setup 
    globals(),locals(),[backend_name]) 
  File 
"/Applications/Spyder.app/Contents/Resources/lib/python2.7/matplotlib/backe nds/backend_qt4agg.py", 
line 13, in \<module> 
    from backend_qt4 import QtCore, QtGui, FigureManagerQT, FigureCanvasQT,\ 
  File 
"/Applications/Spyder.app/Contents/Resources/lib/python2.7/matplotlib/backe nds/backend_qt4.py", 
line 24, in \<module> 
    from qt4_compat import QtCore, QtGui, _getSaveFileName, __version__ 
  File 
"/Applications/Spyder.app/Contents/Resources/lib/python2.7/matplotlib/backe nds/qt4_compat.py", 
line 36, in \<module> 
    import sip 
ImportError: No module named sip

_Original issue: http://code.google.com/p/spyderlib/issues/detail?id=1277_

**Comment by spyder-issues-migrator** [comment]
_From [ccordoba12](https://code.google.com/u/ccordoba12/) on 2013-02-23T07:45:47Z_

I'm sorry for not answering this issue for you on the list. You need to deactivate our PYTHONSTARTUP replacement.

For that you have to go to:

Tools > Preferences > Console > Advanced Settings > PYTHONSTARTUP replacement

and select the option

[ ] Default PYTHONSTARTUP script

Besides, I should mention that if you want to use IPython with Spyder and EPD, you need to update it inside the last one. For that you need to open a terminal and write:

enpkg enstaller
enpkg ipython

and then restart Spyder

**Status:** WontFix

**Comment by spyder-issues-migrator** [comment]
_From [breed...@gmail.com](https://code.google.com/u/105348566576685241380/) on 2013-02-25T08:40:30Z_

Thanks!  This seemed to work on my work computer with the full 64bit version of EPD.  I got some errors on my home computer with EPD_free.  I'll try it again, and then post the error message if it happens again.

**Comment by spyder-issues-migrator** [comment]
_From [breed...@gmail.com](https://code.google.com/u/105348566576685241380/) on 2013-02-27T06:06:07Z_

I actually started having issues with the full version of EPD on my work computer.  I will post the error messages in the next few days.

Here is what I get at the console of my home computer with the free version of EPD:  

Python 2.7.3 |CUSTOM| (default, Apr 12 2012, 11:28:34) 
[GCC 4.0.1 (Apple Inc. build 5493)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
Exception in thread Thread-1:
Traceback (most recent call last):
  File "threading.pyo", line 551, in __bootstrap_inner
  File "/Applications/Spyder.app/Contents/Resources/lib/python2.7/spyderlib/widgets/externalshell/monitor.py", line 568, in run
    already_pickled=True)
  File "/Applications/Spyder.app/Contents/Resources/lib/python2.7/spyderlib/utils/bsdsocket.py", line 29, in write_packet
    sock.send(struct.pack("l", len(sent_data)) + sent_data)
error: [Errno 32] Broken pipe

I am starting to wonder if I should just bite the bullet and install Spyder via Macports.

**Comment by spyder-issues-migrator** [comment]
_From [ccordoba12](https://code.google.com/u/ccordoba12/) on 2013-02-27T06:13:31Z_

I think we fix this one on beta3. Could you try it?

**Comment by spyder-issues-migrator** [comment]
_From [breed...@gmail.com](https://code.google.com/u/105348566576685241380/) on 2013-02-28T16:13:57Z_

Ok I updated to Spyder Version:  2.2.0beta3 and tried it on my work computer with the full 64 bit version of EPD.  It seems to work properly except runfile() (F5 on the keyboard) does not work.  The console says "name 'runfile' is not defined.  

Is runfile() is normally imported using the scientific startup script?  (I used the default PYTHONSTARTUP script like you told me to.)

**Comment by spyder-issues-migrator** [comment]
_From [ccordoba12](https://code.google.com/u/ccordoba12/) on 2013-03-01T08:50:11Z_

You're totally right: runfile is not present when you switch interpreters and hence it can't be used. The scientific startup script only loads Numpy, Scipy and matplotlib when the interpreter is started, so it doesn't have anything to do with runfile.

Please open a new issue to see what we can do it.

**Status:** Invalid

**Comment by spyder-issues-migrator** [comment]
_From [fi...@moosvi.com](https://code.google.com/u/113045289775469152004/) on 2013-03-01T21:36:26Z_

**UPDATE**

Ha, nevermind, I forgot when I reset the spider prefs, python executable defaults back to the OS X installation.

Things still aren't working for me with Enthought's distro and Spyder

**Comment by spyder-issues-migrator** [comment]
_From [fi...@moosvi.com](https://code.google.com/u/113045289775469152004/) on 2013-03-01T21:38:49Z_

The iphthon console does not load on launch - I'm met with a grey screen. The regular python console works fine, but doesn't let me actually run any programs (run file error above)

**Comment by spyder-issues-migrator** [comment]
_From [stefan.r...@gmail.com](https://code.google.com/u/111201285089679297844/) on 2013-03-01T21:47:31Z_

I'm not clear I'm following what your problems are but I seem to have related errors:
I have EPD 7.3 with upgraded ipython, I have set the spyder python executable to the EPD one and when I start spyder I see this message in the internal console:

Traceback (most recent call last):
  File "/Applications/Spyder.app/Contents/Resources/lib/python2.7/spyderlib/widgets/externalshell/start_ipython_kernel.py", line 170, in \<module>
    ipk_temp.config = kernel_config()
  File "/Applications/Spyder.app/Contents/Resources/lib/python2.7/spyderlib/widgets/externalshell/start_ipython_kernel.py", line 36, in kernel_config
    from spyderlib.config import CONF
ImportError: No module named spyderlib.config

Furthermore, the standard console that greets me one startup is this one:

Python 2.7.3 |CUSTOM| (default, Apr 12 2012, 11:14:05) 
[GCC 4.0.1 (Apple Inc. build 5493)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
\>>> 

I would have expected to see the EPD python (that I do see when I start python, or ipython) on the commandline. Can we re-open this issue? 
Anyone with a working EPD and spyder running in harmony, please feel free to let us know that we are aiming for something doable.

**Comment by spyder-issues-migrator** [comment]
_From [ccordoba12](https://code.google.com/u/ccordoba12/) on 2013-03-05T09:53:04Z_

Ok, there are three issues here:

1. Our scientific startup script has to be disabled when switching to EPD.
2. runfile is not working after the switch, so you can't run files.
3. IPython consoles are not working now (they were in beta1, I think).

I'll try to solve them for your next release. 1. and 3. shouldn't be so hard, but 2. could be tricky.

@-Stefan: That |CUSTOM| on the line

Python 2.7.3 |CUSTOM| (default, Apr 12 2012, 11:14:05)

comes from EPD, so you are already using it. I'll see what I can do to make it print the right thing, i.e.

|EPD_free 7.3-1 (64-bit)|

**Summary:** Enthought Python Distribution and Spyder DMG are not working well on Mac OS X  \<span class="oldvalue"> (was: Enthought Python Distribution and Spyder DMG installation on Mac OS X) \</span>  
  **Status:** Accepted  
  **Labels:** MS-v2.2 OpSys-OSX

**Comment by spyder-issues-migrator** [comment]
_From [stefan.r...@gmail.com](https://code.google.com/u/111201285089679297844/) on 2013-03-05T22:06:44Z_

Just for the record. Both EPD MacOSX 32bit and 64bit have similar kind of problems. There are some minor though probably irrelevant differences between the two packages such as wxpython in the 32bit see here: https://www.enthought.com/repo/epd/installers/ "Special note about 64-bit Mac OSX
The macosx-x86_64 installer does not include wxPython, which a lot of ETS relies on. Even though ETS supports the use of PySide as a backend, things are much less stable under Qt. Therefore, if you want to use ETS, we recommned using the 32-bit EPD on Mac OSX. "

**Comment by spyder-issues-migrator** [comment]
_From [ccordoba12](https://code.google.com/u/ccordoba12/) on 2013-03-09T20:08:16Z_

This issue was updated by revision e2e335803d02 .


- Previous work (the one that solved issue #1226 ) was a hack that was not
working well.
- Use our sitecustomize script with external interpreters so that runfile
and debugfile are present in them.
- Add the app's main python path to the end of sys.path so that IPython
consoles start and work as expected (if the external interpreter has it
installed).

**Status:** Started

**Comment by spyder-issues-migrator** [comment]
_From [ccordoba12](https://code.google.com/u/ccordoba12/) on 2013-03-09T20:08:16Z_

This issue was updated by revision e5d2c92ca0db .

**Comment by spyder-issues-migrator** [comment]
_From [ccordoba12](https://code.google.com/u/ccordoba12/) on 2013-03-09T20:08:17Z_

This issue was updated by revision aebf0789020d .

**Comment by spyder-issues-migrator** [comment]
_From [ccordoba12](https://code.google.com/u/ccordoba12/) on 2013-03-12T11:39:06Z_

I uploaded a new DMG (beta4), please test it and tell me if things work for you. All three problems in comment `#13` should be fixed.

**Comment by spyder-issues-migrator** [comment]
_From [fi...@moosvi.com](https://code.google.com/u/113045289775469152004/) on 2013-03-12T11:48:00Z_

YES!!! This works for me now!

more specifically, with Spyder I can now use ipython with EPD64, the python interpreter with EPD64, and use the default scientific startup script!

Thanks Carlos!

**Attachment:** [Screen Shot 2013-03-12 at 11.47.50 AM.png](http://code.google.com/p/spyderlib/issues/detail?id=1277#hc20)

**Comment by spyder-issues-migrator** [comment]
_From [breed...@gmail.com](https://code.google.com/u/105348566576685241380/) on 2013-03-13T08:22:48Z_

At first glance it appears the three problems in comment `#13` are fixed.  Thanks, Carlos.

However, the PYTHONPATH manager does not seem to be working with the EPD64 interpreter.  When I do "print sys.path" with the default python interpreter, the directories listed in the PYTHONPATH manager show up.  When I do "print sys.path" with the EPD64 interpreter, the directories listed in the PYTHONPATH manager do not show up.

**Comment by spyder-issues-migrator** [comment]
_From [stefan.r...@gmail.com](https://code.google.com/u/111201285089679297844/) on 2013-03-13T10:45:42Z_

me too: Thank you Carlos and the PYTHONPATH manager settings appear to be ignored in the EPD python interpreter. However, EPD python does seem to honour the $PYTHONPATH environment variable so it's not a big problem.

**Comment by spyder-issues-migrator** [comment]
_From [ccordoba12](https://code.google.com/u/ccordoba12/) on 2013-03-14T05:51:34Z_

Ok, thanks a lot for the confirmation to the three of you. I'd say this is verified! :)

@-breedlun or @-stefan: Please open a new issue for our PYTHONPATH manager not working with EPD. I didn't test it so I didn't know it wasn't working. However, since it's a minor thing compared to the others, I'll fix it after 2.2 is released.

**Status:** Verified

**Comment by spyder-issues-migrator** [comment]
_From [breed...@gmail.com](https://code.google.com/u/105348566576685241380/) on 2013-03-17T08:15:18Z_

It seems that Spyder 2.2.0beta4 does not play well with the free version of EPD on my home computer.  (In comment `#21` above, I was referring to my work computer, which has the full version of EPD64.)  I set Spyder 2.2.0beta4's interpreter to /Library/Frameworks/Python.framework/Versions/7.3/bin/python2.7, and I got the error list below in the console.  I should also note that my work computer is running OS X 10.7 (Lion), while my home computer is running OS X 10.8 (Mountain Lion). 

Spyder 2.2.0beta4 internal shell on Python 2.7.3 64bits [Darwin]
Inspect Spyder internals:
  spy.app, spy.window, dir(spy)
\>>> Exception in thread Thread-3:
Traceback (most recent call last):
  File "threading.pyc", line 551, in __bootstrap_inner
  File "/Applications/Spyder.app/Contents/Resources/lib/python2.7/spyderlib/widgets/externalshell/introspection.py", line 71, in run
  File "/Applications/Spyder.app/Contents/Resources/lib/python2.7/spyderlib/utils/bsdsocket.py", line 54, in read_packet
OverflowError: signed integer is greater than maximum

Exception in thread Thread-2:
Traceback (most recent call last):
  File "threading.pyc", line 551, in __bootstrap_inner
  File "/Applications/Spyder.app/Contents/Resources/lib/python2.7/spyderlib/widgets/externalshell/introspection.py", line 71, in run
  File "/Applications/Spyder.app/Contents/Resources/lib/python2.7/spyderlib/utils/bsdsocket.py", line 54, in read_packet
OverflowError: signed integer is greater than maximum

**Comment by spyder-issues-migrator** [comment]
_From [ccordoba12](https://code.google.com/u/ccordoba12/) on 2013-03-17T08:15:29Z_

Are you using EPD 32 or 64 bits on your home computer?

**Comment by spyder-issues-migrator** [comment]
_From [breed...@gmail.com](https://code.google.com/u/105348566576685241380/) on 2013-03-17T08:26:41Z_

Based on this web page: http://www.enthought.com/products/epd_free.php I believe I am using EPD-free 32-bit, since that is the only version available for macs.

**Comment by spyder-issues-migrator** [comment]
_From [ccordoba12](https://code.google.com/u/ccordoba12/) on 2013-03-17T08:41:40Z_

I think you are, yep. Those OverflowError's seem to be due to a mismatch in between 32 and 64 bits.

This represents a real problem, because the App is currently built for 64 bits and I don't have the resources nor the time to build it for two archs right now...

**Comment by spyder-issues-migrator** [comment]
_From [ccordoba12](https://code.google.com/u/ccordoba12/) on 2013-03-17T08:45:03Z_

Would you (or someone else in this issue) be able to help me? I can give you detailed instructions on how to do build the App.

**Comment by spyder-issues-migrator** [comment]
_From [breed...@gmail.com](https://code.google.com/u/105348566576685241380/) on 2013-03-17T08:52:47Z_

I would like to, but it sounds like it will take some time and effort.  I am no expert at the command line, and I am already over committed right now.  Sorry.

In the future, when I am less busy, I will be willing to contribute.

**Comment by spyder-issues-migrator** [comment]
_From [ccordoba12](https://code.google.com/u/ccordoba12/) on 2013-03-17T09:35:52Z_

Don't worry about it, it's tricky and it really takes some time.

**Comment by spyder-issues-migrator** [comment]
_From [ccordoba12](https://code.google.com/u/ccordoba12/) on 2013-04-06T20:29:06Z_

**Labels:** Cat-Console

**Comment by Stretch97** [comment]
I thought I would post an update on this issue.  Now that Enthought provides a free version of 64bit Canopy, I installed it, and downloaded the DMG for Spyder 2.3.2.  I fired up Spyder, pointed the "Path to Python Interpreter Executable Binary" to ~/Library/Enthought/Canopy_64bit/User/bin/python, and turned off the scientific startup script.  I restarted Spyder and the Console window just stays blank.  No error or anything.

P.S. As a workaround, I have successfully built spyder from source with Canopy 64bit, and everything (except for the profiler) seems to work correctly.  Thus, I don't need this issue solved right away.
